### ISSUE TITLE
Option to disable key-spacing for singleLine

### ISSUE THREAD

**Issue by mnpenner** [eslint/eslint#7318]
**What rule do you want to change?**

`key-spacing`

**Does this change cause the rule to produce more or fewer warnings?**

Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**

The `singleLine` and `multiLine` options would gain a new valid setting, `"off"` (or alternatively, `false`). e.g.

```
"key-spacing": ["error", {
    singleLine: "off",
    multiLine: {
        beforeColon: false,
        afterColon: true,
        mode: 'strict',
    }
}],
```

**Please provide some example code that this change will affect:**

This would disable the inspection for single-line objects:

``` jsx
// correct
<BlueBox className="cr-form" style={{minWidth:'500px'}}>

// correct
<BlueBox className="cr-form" style={{minWidth: '500px'}}>
```

While still enforcing spacing on multi-line objects such as:

``` js
// correct
call({
    foobar: 42,
    bat: 2 * 2
});

// incorrect
call({
    foobar:42,
    bat:2 * 2
});
```

**What does the rule currently do for this code?**

Prevents spaceless single-line objects from being valid.

**What will the rule do after it's changed?**

Allow you to enable/disable `singleLine` and `multiLine` settings independently.

---

Alternatively, if we could specify that `0` _or_ `1` space is allowed after the colon, that would also work even better than disabling it completely. You could implement this as `mode: "loose"`.

Version: 3.7.1

**Comment by not-an-aardvark** [comment]
Thanks for the feature request. To make it easier for other people who are going to look at it, would you mind filling out the [rule change proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md)?

**Comment by eslintbot** [comment]
Hi @mnpenner, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by mnpenner** [comment]
@not-an-aardvark Sorry, I somehow missed that template. I've edited the issue description to use the correct template. Thanks!

**Comment by not-an-aardvark** [comment]
Thanks for filling in the info.

I suppose being able to disable checking for a particular option could be useful. However, I think passing `null` would be better than `false` or `"off"`.

**Comment by mnpenner** [comment]
@not-an-aardvark That's fine by me. I only chose "off" because the root-level options use "off".

**Comment by btmills** [comment]
I like that "off" is explicit. I'll give a :+1: for it. @mnpenner would you be willing to send a PR to implement this if it's accepted?

**Comment by mnpenner** [comment]
Uhh...I suppose I can take a crack at it. Might take me awhile to get to it though. Do you guys prefer `"off"` or `null`?

**Comment by kaicataldo** [comment]
Looks like this has three :+1: from the team, but we do need a champion to drive this forward. @eslint/eslint-team Would anyone like to step up?

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
